### PR TITLE
Keep names of all products + new BinaryProduct

### PR DESCRIPTION
**Pull request by dsavchenko**

### CODE REVIEW

**Comment by burnout87** [comment]
Live test is failing in the same way as https://github.com/oda-hub/oda_api/pull/199#issuecomment-1604285566

File: oda_api/api.py [hunk 1]
```diff
@@ -1250,28 +1256,38 @@ def from_response_json(cls, res_json, instrument, product):
...
 
         if 'binary_data_product_list' in res_json['products'].keys():
-            data.extend([BinaryData().decode(d)
-                         for d in res_json['products']['binary_data_product_list']])
+            try:
+                data.extend([BinaryProduct.decode(d)
+                             for d in res_json['products']['binary_data_product_list']])
+            except:
+                data.extend([BinaryData().decode(d)
+                             for d in res_json['products']['binary_data_product_list']])
```
Comment (burnout87) [inline oda_api/api.py:1264]: why is this needed?

Comment (dsavchenko) [reply oda_api/api.py:1264]: The except case is just for backward compatibility, New BinaryProduct will be used in general

File: oda_api/data_products.py [hunk 2]
```diff
@@ -240,9 +276,9 @@ def _check_dict(self,_kw):
 
 
     @classmethod
-    def from_fits_hdu(cls,hdu,name=''):
+    def from_fits_hdu(cls,hdu,name=None):
 
-        if name=='':
+        if not name:
```
Comment (burnout87) [inline oda_api/data_products.py:281]: I think PEP8 recommends to always use `is` operator for checking against `None`

**Review by burnout87 — COMMENTED** [review]
few  comments/questions

File: oda_api/data_products.py (hunk 2 above)
Comment (dsavchenko) [reply oda_api/data_products.py:281]: Here I wanted to account for both None and empty string (compatibility again, but not sure if it's really needed)

Comment (burnout87) [reply oda_api/data_products.py:281]: then  I'd put:

```python
if name is not None or name == '':
```

**Review by burnout87 — DISMISSED** [review]

File: oda_api/api.py [hunk 3]
```diff
@@ -1250,28 +1257,36 @@ def from_response_json(cls, res_json, instrument, product):
                          for d in res_json['products']['numpy_data_product_list']])
 
         if 'binary_data_product_list' in res_json['products'].keys():
-            data.extend([BinaryData().decode(d)
-                         for d in res_json['products']['binary_data_product_list']])
+            try:
+                data.extend([BinaryProduct.decode(d)
+                             for d in res_json['products']['binary_data_product_list']])
+            except:
```
Comment (volodymyrss) [inline oda_api/api.py:1263]: could you catch something more specific?

**Comment by volodymyrss** [comment]
what's with Untitled.ipynb

**Comment by dsavchenko** [comment]
> what's with Untitled.ipynb

It was added by mistake

**Review by volodymyrss — APPROVED** [review]
